src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -420,31 +420,32 @@ private async Task StartAsyncCore(CancellationToken cancellationToken)
             CheckDisposed();
 
             Log.Starting(_logger);
-
-            // Start the connection
-            var connection = await _connectionFactory.ConnectAsync(_protocol.TransferFormat);
-            var startingConnectionState = new ConnectionState(connection, this);
-
-            // From here on, if an error occurs we need to shut down the connection because
-            // we still own it.
-            try
-            {
-                Log.HubProtocol(_logger, _protocol.Name, _protocol.Version);
-                await HandshakeAsync(startingConnectionState, cancellationToken);
-            }
-            catch (Exception ex)
+            using (var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _state.StopCts.Token))
             {
-                Log.ErrorStartingConnection(_logger, ex);
+                // Start the connection
+                var connection = await _connectionFactory.ConnectAsync(_protocol.TransferFormat, cts.Token);
+                var startingConnectionState = new ConnectionState(connection, this);
 
-                // Can't have any invocations to cancel, we're in the lock.
-                await CloseAsync(startingConnectionState.Connection);
-                throw;
-            }
+                // From here on, if an error occurs we need to shut down the connection because
+                // we still own it.
+                try
+                {
+                    Log.HubProtocol(_logger, _protocol.Name, _protocol.Version);
+                    await HandshakeAsync(startingConnectionState, cancellationToken);   //here we use cancellationToken, as HandshakeAsync combines cts for itself.
+                }
+                catch (Exception ex)
+                {
+                    Log.ErrorStartingConnection(_logger, ex);
 
-            // Set this at the end to avoid setting internal state until the connection is real
-            _state.CurrentConnectionStateUnsynchronized = startingConnectionState;
-            startingConnectionState.ReceiveTask = ReceiveLoop(startingConnectionState);
+                    // Can't have any invocations to cancel, we're in the lock.
+                    await CloseAsync(startingConnectionState.Connection);
+                    throw;
+                }
 
+                // Set this at the end to avoid setting internal state until the connection is real
+                _state.CurrentConnectionStateUnsynchronized = startingConnectionState;
+                startingConnectionState.ReceiveTask = ReceiveLoop(startingConnectionState);
+            }
```

Comment:
Nit: newline after the end of blocks

Usually we  also put newlines before the start of a block too unless we just entered a new block or acquired a resource to only be used in the block (e.g. acquiring the connection lock immediately before a try/finally).

Reply:
OK. Fixed.